hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightSearchEndpoint.java

```diff
@@ -0,0 +1,427 @@
+        Table.KeyValue<String, OmKeyInfo> entry = keyIter.next();
+        String dbKey = entry.getKey();
+        if (!dbKey.startsWith(startPrefix)) {
+          break; // Exit the loop if the key no longer matches the prefix
+        }
+        matchedKeys.put(dbKey, entry.getValue());
+      }
+    } catch (IOException exception) {
+      LOG.error("Error retrieving keys from table for path: {}", startPrefix,
+          exception);
+      throw exception;
+    }
+    return matchedKeys;
+  }
+
+  /**
+   * Creates a KeyEntityInfo object from an OmKeyInfo object and the corresponding key.
+   *
+   * @param dbKey   The key in the database corresponding to the OmKeyInfo object.
+   * @param keyInfo The OmKeyInfo object to create the KeyEntityInfo from.
+   * @return The KeyEntityInfo object created from the OmKeyInfo object and the key.
+   */
+  private KeyEntityInfo createKeyEntityInfoFromOmKeyInfo(String dbKey,
+                                                         OmKeyInfo keyInfo) {
+    KeyEntityInfo keyEntityInfo = new KeyEntityInfo();
+    keyEntityInfo.setKey(dbKey); // Set the DB key
+    keyEntityInfo.setPath(
+        keyInfo.getKeyName()); // Assuming path is the same as key name
+    keyEntityInfo.setInStateSince(keyInfo.getCreationTime());
+    keyEntityInfo.setSize(keyInfo.getDataSize());
+    keyEntityInfo.setReplicatedSize(keyInfo.getReplicatedSize());
+    keyEntityInfo.setReplicationConfig(keyInfo.getReplicationConfig());
+    return keyEntityInfo;
+  }
+
+  /**
+   * Constructs an object path with the given IDs.
+   *
+   * @param ids The IDs to construct the object path with.
+   * @return The constructed object path.
+   */
+  private String constructObjectPathWithPrefix(long... ids) {
+    StringBuilder pathBuilder = new StringBuilder();
+    for (long id : ids) {
+      pathBuilder.append(OM_KEY_PREFIX).append(id);
+    }
+    return pathBuilder.toString();
+  }
+
+  /**
+   * Returns a response indicating that no keys matched the search prefix.
+   *
+   * @param startPrefix The search prefix that was used.
+   * @return The response indicating that no keys matched the search prefix.
+   */
+  private Response noMatchedKeysResponse(String startPrefix) {
+    String jsonResponse = String.format(
+        "{\"message\": \"No keys matched the search prefix: '%s'.\"}",
+        startPrefix);
+    return Response.status(Response.Status.NOT_FOUND)
+        .entity(jsonResponse)
+        .type(MediaType.APPLICATION_JSON)
+        .build();
+  }
+
+  /**
+   * Utility method to create a bad request response with a custom message.
+   * Which means the request sent by the client to the server is incorrect
+   * or malformed and cannot be processed by the server.
+   *
+   * @param message The message to include in the response body.
+   * @return A Response object configured with the provided message.
+   */
+  private Response createBadRequestResponse(String message) {
```

Comment:
Instead of calling this method for negative limit, use it to validate proper path. Backend API should never be dependent on assumption that UI will pass correct path. It should be resilient with outside client like curl or postman as well. Client should not pass some junk characters in path, so validate that and use this method to send `BAD_REQUEST` response.

Reply:
It will now validate the names of volumes and buckets